File: src/main/java/at/hannibal2/skyhanni/utils/jsonobjects/MayorJson.java
package at.hannibal2.skyhanni.utils.jsonobjects;

import com.google.gson.annotations.Expose;

import java.util.ArrayList;

public class MayorJson {
    @Expose
    public boolean success;
    @Expose
    public long lastUpdated;
    @Expose
    public Mayor mayor;
    @Expose
    public Election current;

    public class Candidate {
        @Expose
        public String key;
        @Expose
        public String name;
        @Expose
        public ArrayList<Perk> perks;
        @Expose
        public int votes;

        @Override
        public String toString() {
            return "Candidate{" +
                    "key='" + key + '\'' +
                    ", name='" + name + '\'' +
                    ", perks=" + perks +
                    ", votes=" + votes +
                    '}';
        }
    }

    public class Election {
        @Expose
        public int year;
        @Expose
        public ArrayList<Candidate> candidates;
    }

    public class Mayor {
        @Expose
        public String key;
        @Expose
        public String name;
        @Expose
        public ArrayList<Perk> perks;
        @Expose
        public Election election;
    }

    public static class Perk {
        @Expose
        public String name;
        @Expose
        public String description;

        @Override
        public String toString() {
            return "Perk{" +
                    "name='" + name + '\'' +
                    ", description='" + description + '\'' +
                    '}';
        }
    }
}

File: src/main/java/at/hannibal2/skyhanni/data/jsonobjects/other/MayorJson.kt
package at.hannibal2.skyhanni.data.jsonobjects.other

import com.google.gson.annotations.Expose

data class MayorJson(
    @Expose val mayor: MayorInfo,
    @Expose val current: MayorElection?,
)

data class MayorInfo(
    @Expose val key: String,
    @Expose val name: String,
    @Expose val perks: List<MayorPerk>,
    // Ministers won't exist,
    // when the current mayor is a special mayor
    @Expose val minister: Minister?,
    @Expose val election: MayorElection,
)

data class MayorElection(
    @Expose val year: Int,
    @Expose val candidates: List<MayorCandidate>,
)

data class MayorCandidate(
    @Expose val key: String,
    @Expose val name: String,
    @Expose val perks: List<MayorPerk>,
    @Expose val votes: Int,
)

data class Minister(
    @Expose val key: String,
    @Expose val name: String,
    @Expose val perk: MayorPerk,
)

data class MayorPerk(
    @Expose val name: String,
    @Expose val description: String,
    @Expose val minister: Boolean = false,
)
